src/dashboards/selectors/index.ts

```diff
@@ -28,12 +28,39 @@ export const getTimeRange = (state: AppState): TimeRange => {
   return state.ranges[contextID] || DEFAULT_TIME_RANGE
 }
 
+/**
+ *  is the string a valid UTC time without any time zone information?
+ *  ie:  "2021-07-17T14:00:00.000Z" is valid
+ *  if it has any timezone offset, then it is not a valid UTC time string
+ *
+ * @param str
+ */
+function isUtcTime(str) {
+  const regex = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z$/
+  return regex.test(str)
+}
+
+/** is the time already in utc format?  if so it does not need converting. */
+const timesNeedConverting = newTimeRange => {
+  // if the time range is *already* utc, does not need converting.  check just the first part, as
+  // they will be formatted the same
+  return !isUtcTime(newTimeRange.lower)
```

Comment:
let's move these into `src/shared/utils/datetime` since they're date/time utils. not sure how we should name the file they belong in, any ideas?

also how about some test cases to try to catch some edge cases?

Reply:
dateTimeUtils.ts should be fine.  

there is already unit tests for the conversion function.  

i can add to make sure the case with custom === true and needs conversion (is both true and false) are in the unit tests.